File: additional_02_tcp_socket/TcpSocketServer.java
// Additional Experiment 2: Basic TCP Socket Server

import java.io.*;
import java.net.*;

public class TcpSocketServer {
    public static void main(String[] args) {
        final int PORT = 6000;

        try (
            ServerSocket server = new ServerSocket(PORT) // Create listening socket.
        ) {
            System.out.println("Waiting for a TCP client...");

            try (
                Socket socket = server.accept();         // Accept client.
                BufferedReader input = new BufferedReader(
                    new InputStreamReader(socket.getInputStream())
                );
                PrintWriter output = new PrintWriter(
                    socket.getOutputStream(), true
                )
            ) {
                String message = input.readLine();        // Receive one message.
                System.out.println("Received: " + message);

                output.println("Message received successfully."); // Send response.
            }
        } catch (IOException error) {
            System.out.println("Server error: " + error.getMessage());
        }
    }
}

File: additional_02_tcp_socket/tcp_socket_server.py
# Additional Experiment 2: Basic TCP Socket Server

import socket

HOST = "0.0.0.0"
PORT = 6000

server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)  # Create TCP socket.
server.bind((HOST, PORT))                                  # Attach socket to port.
server.listen(1)                                           # Listen for one client.

print("Waiting for a TCP client...")
connection, address = server.accept()                      # Accept connection.
print("Connected:", address)

data = connection.recv(1024).decode()                      # Receive one message.
print("Received:", data)

connection.sendall(b"Message received successfully.")       # Send response.

connection.close()                                         # Close client connection.
server.close()                                              # Close server socket.
